src/main/java/com/dev/cinema/controllers/OrderController.java

```diff
@@ -33,16 +34,18 @@ public OrderController(OrderService orderService, OrderMapper orderMapper,
     }
 
     @PostMapping("/complete")
-    public void complete(@RequestParam Long userId) {
-        User user = userService.getById(userId);
+    public void complete(Authentication authentication) {
+        String email = ((UserDetails) authentication.getPrincipal()).getUsername();
```

Comment:
try to use authentication.getName() instead

Reply:
done